content/slides/data-inspection.qmd

```diff
@@ -18,35 +18,383 @@ the same datum using e.g. `gdalwarp`, `ogr2ogr`.
+    source=Mass conservation (Mathieu Morlighem)
+    standard_name=bedrock_altitude
+    STATISTICS_MAXIMUM=3672.35546875
+    STATISTICS_MEAN=-307.31420082049
+    STATISTICS_MINIMUM=-5571.6728515625
+    STATISTICS_STDDEV=1160.9999064556
+    STATISTICS_VALID_PERCENT=57.73
+    units=meters
+    _FillValue=-9999
+```
 
-...
+::: {.notes}
+* Grid size, in "pixels"
+* Well-Known Text represents CRS with hierarchical data, which `gdalinfo`
+nicely indents for us. The "top-level" EPSG is shown as 3413.
+* Pixel size, in projected units
+* Global metadata: _TODO: Mention of the difference namespaces, specifically
+  call out the `grid_mapping` namespace's significance?_
+* Band metadata
+* Overviews: Various down-sampled versions of the data for performance when
+  zoomed out.
+:::
+
+
+## Raster data inspection: Rasterio CLI
+
+`rio info --help`
 
+```{.text code-line-numbers="false"}
+$ rio info --crs bedmachine_bed.tif
+EPSG:3413
+
+$ rio info --dtype bedmachine_bed.tif
+float32
+
+$ rio info --nodata bedmachine_bed.tif
+-9999.0
+
+$ rio info --shape bedmachine_bed.tif
+21876 14844
+```
+
+
+## Raster data inspection: Rasterio Python
+
+`import rasterio; help(rasterio)`
+
+```{.text code-line-numbers="false"}
+$ python
+Python 3.11.0 | packaged by conda-forge | (main, Jan 14 2023, 12:27:40) [GCC 11.3.0] on linux
+Type "help", "copyright", "credits" or "license" for more information.
+>>> import rasterio
+>>> ds = rasterio.open('bedmachine_bed.tif')
+>>> ds.crs
+CRS.from_epsg(3413)
+>>> ds.dtypes
+('float32',)
+>>> ds.nodata
+-9999.0
+>>> ds.shape
+(21876, 14844)
+>>> ds.close()
+```
+
+::: {.notes}
+Normally, you would use a context manager to open a file with rasterio. Since
+we didn't do that here, we must manually close the file.
+:::
 
-## Raster data inspection with Rasterio
 
-...
 
 
 ## Raster data inspection with Xarray (rioxarray)
 
-...
+`import xarray; help(xarray)`
+
+```{.text code-line-numbers="false"}
+>>> import rioxarray
+>>> import xarray as xra
+>>> ds = xra.open_dataset('bedmachine_bed.tif')
+>>> ds
+<xarray.Dataset>
+Dimensions:  (band: 1, x: 14844, y: 21876)
+Coordinates:
+  * band     (band) int64 1
+  * x        (x) float64 -9.979e+05 -9.978e+05 ... 1.228e+06 1.228e+06
+  * y        (y) float64 -3.298e+05 -3.3e+05 ... -3.611e+06 -3.611e+06
+    mapping  int64 ...
+Data variables:
+    bed      (band, y, x) float32 ...
+>>> ds.rio.crs
+CRS.from_epsg(3413)
+```
+
+::: {.notes}
+`rioxarray` provides geospatial awareness to `xarray`.
+:::
 
 
 # Vector data inspection tools
 
-## Vector data inspection with OGR
+## Vector data inspection: `ogrinfo`
+
+* <https://gdal.org/programs/ogrinfo.html>
+* `ogrinfo --help`
+* `man ogrinfo`
 
-...
 
+## Vector data inspection: `ogrinfo`
 
-## Vector data inspection with Fiona
+```{.text code-line-numbers="1|2-3|5|6|7|8|9|56|58-86"}
+$ gdalinfo -al -so populated_places.gpkg
+INFO: Open of `populated_places.gpkg'
+      using driver `GPKG' successful.
 
-...
+Layer name: SELECT
```

Comment:
Looks like this layer name is "SELECT" because we use SQL in our `ogr2og4` step. I wonder if we could use an `as placenames` or something to set the name more appropriately

Reply:
Makes sense. If that doesn't work, there must be an `ogr2ogr` option we could use! Maybe a layer creation option (`-lco`)